### PR TITLE
Add built-in dict.pop

### PR DESCRIPTION
**Pull request by youknowone**

### CODE REVIEW

File: vm/src/obj/objdict.rs [hunk 1]
```diff
@@ -228,6 +228,13 @@ impl PyDictRef {
         PyDictRef::merge(&self.entries, dict_obj, kwargs, vm)
     }
 
+    fn pop(self, key: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        match self.entries.borrow_mut().pop(vm, &key) {
```
Comment (windelbouwman) [inline vm/src/obj/objdict.rs]: You may want to use `map_err` here. Furthermore, I think we do not require the map the result. Can't we do without this match statement and return whatever the `pop` function on entries returns?

File: vm/src/obj/objdict.rs [hunk 2]
```diff
@@ -228,6 +228,13 @@ impl PyDictRef {
...
+            Ok(obj) => Ok(obj),
+            Err(_) => Err(vm.new_key_error(format!("{}", vm.to_pystr(&key)?))),
```
Comment (windelbouwman) [inline vm/src/obj/objdict.rs]: I would move the creation of this key error to line 207 of dictdatatype.rs.

**Comment by windelbouwman** [comment]
@youknowone thank you for your contribution. I left some remarks, but the change looks good to me! Let me know if you want to modify the changes or not, then we can merge this work.

**Comment by youknowone** [comment]
Thanks for the nice review. I fixed it.

**Pull request merged**